### PR TITLE
Made `numba.numpy_support` available without an import

### PR DESCRIPTION
**Pull request by eric-wieser**
Closes https://github.com/numba/numba/issues/5466

### CODE REVIEW

File: numba/__init__.py [hunk 1]
```diff
@@ -58,6 +58,31 @@
...
+            pass
+        else:
+            errors.deprecate_moved_module(submodule_name, new_name)
+            return importlib.import_module(submodule_name)
+
+        # produce the normal error message
+        from types import ModuleType
+        ModuleType.__getattribute__(sys.modules[__name__], attr)
+
+
```
Comment (stuartarchibald) [inline numba/__init__.py:85]: Should this be rolled into https://github.com/numba/numba/pull/5498 ?

Comment (eric-wieser) [reply numba/__init__.py:85]: I'll rebase it onto #5498 once that gets merged.

Comment (stuartarchibald) [reply numba/__init__.py:85]: Sounds good, thanks.

File: numba/__init__.py [hunk 2]
```diff
@@ -58,6 +58,31 @@
 test = runtests.main
 
 
+# needed for compatibility until 0.50.0. Note that accessing members of
+# numpy_support will raise DeprecationWarning
+_auto_import_submodules = {
+    'numba.numpy_support': 'numba.core.numpy_support'
```
Comment (eric-wieser) [inline numba/__init__.py:64]:
```suggestion
    'numba.numpy_support': 'numba.core.numpy_support'
```
```suggestion
    'numba.targets': None
```

**Comment by sklam** [comment]
I have extended this PR to cover all other `numba.<mod>` in https://github.com/numba/numba/pull/5532

**Pull request closed without merging**